### PR TITLE
fix: resolved failing of continue compilation flag for error files for different kinds

### PR DESCRIPTION
**Pull request by parth121101**
Towards: #5463 #5036

### CODE REVIEW

File: src/lfortran/semantics/ast_common_visitor.h [hunk 1]
```diff
@@ -3064,9 +3064,15 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                         s_intent == ASRUtils::intent_inout));
                 }
                 ASR::symbol_t *type_declaration;
-                ASR::ttype_t *type = determine_type(x.base.base.loc, sym, x.m_vartype, is_pointer,
+                ASR::ttype_t *type = nullptr;
+                try {
+                type = determine_type(x.base.base.loc, sym, x.m_vartype, is_pointer,
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:3069]: I think the indentation here is incorrect?

**Review by certik — COMMENTED** [review]
I think that this is fine. I am not super happy for adding all these try/catch everywhere --- is there a way to catch it at an upper level?

**Comment by parth121101** [comment]
> I think that this is fine. I am not super happy for adding all these try/catch everywhere --- is there a way to catch it at an upper level?

Will figure this out @certik

**Comment by Pranavchiku** [comment]
> is there a way to catch it at an upper level?

Yes, let's catch it inside `ast_symboltable_visitor` where we run a loop and  `visit_unit_decl**`  for each `unit`

**Review by certik — APPROVED** [review]
Thanks!

**Pull request merged**